### ISSUE TITLE
Incorrect assumption about signature length

### ISSUE THREAD

**Issue by moreal** [odanado/aws-kms-provider#281]
Hello, I'm using this library in my project. 🙏🏻 

There is a bug to assume the length of signature's `r` and `s` as only `32` bytes. I have ever seen 31 bytes, 32 bytes, 33 bytes 🤔 When there is the negative `r` or `s`, 33 bytes, `asn1-parser/parseSignature` slices it until 32 bytes so it become polluted invalid signature.

If you assumed it with some documentation, could you share it?

**Comment by odanado** [comment]
@moreal 
Thank you for the report.

I didn't know about that bug.
I would like to know the details of the bug and can you provide the minimum configuration in which the bug is reproduced?

**Comment by moreal** [comment]
The error is thrown from `Signature.fromDigest` method so I prepared its arguments. I got the arguments by amending the code like below:

```
// const digest = Buffer.from("A6016B5462ACEB9015624966DFA14FFF2048C44653C18C8C8E0F806C5F885D9C", "hex");
async sign(digest) {
        const asn1Signature = await this._sign(digest);
        const address = await this.getAddress();
        const { r, s } = asn1_parser_1.parseSignature(asn1Signature);
        console.log("asn1Signature", asn1Signature.toString("hex"));
        console.log("address", address.toString());
        console.log("r", r.toString("hex"));
        console.log("s", s.toString("hex"))
        return signature_1.Signature.fromDigest(digest, address, r, s);
    }
```

And the log lines were:

```
asn1Signature 30440221009c68bf9b88814142fef77d95956b21625789c34fde9b853653b24fc23515577f021f74e3e4d71e7385ae71042b0f99f7fbbf66e7760dd513ed2fcea754e2a9131c
address b4438C3d302C7c0E41c4c7cc1DFcB17a1F93096b
r 9c68bf9b88814142fef77d95956b21625789c34fde9b853653b24fc23515577f
s 74e3e4d71e7385ae71042b0f99f7fbbf66e7760dd513ed2fcea754e2a9131c
```

**Comment by moreal** [comment]
It seems a bug occurred from unpadded bytes (i.e. no care about 31 bytes case).

So I tried to fix the bug in #283.
